src/core/pointcloud/qgspointcloudrenderer.h

```diff
@@ -570,6 +587,42 @@ class CORE_EXPORT QgsPointCloudRenderer
      */
     virtual QStringList legendRuleKeys() const;
 
+    /**
+     * Sets whether eye dome lighting effect will be used
+     * \since QGIS 3.28
+     */
+    bool useEyeDomeLighting() const { return mUseEyeDomeLighting; }
+
+    /**
+     * Returns whether eye dome lighting effect will be used
+     * \since QGIS 3.28
+     */
+    void setUseEyeDomeLighting( bool useEyeDomeLighting ) { mUseEyeDomeLighting = useEyeDomeLighting; }
+
+    /**
+     * Returns the eye dome lighting strength value
+     * \since QGIS 3.28
+     */
+    double eyeDomeLightingStrength() const { return mEyeDomeLightingStrength; }
+
+    /**
+     * Sets the eye dome lighting strength value
+     * \since QGIS 3.28
+     */
+    void setEyeDomeLightingStrength( double strength ) { mEyeDomeLightingStrength = strength; }
+
+    /**
+     * Returns the eye dome lighting distance value
+     * \since QGIS 3.28
+     */
+    int eyeDomeLightingDistance() const { return mEyeDomeLightingDistance; }
```

Comment:
What's the units here? It would be better to make this a double and add a corresponding QgsUnitTypes::RenderUnit parameter, and then convert to painter units using that. Otherwise we won't get consistent results from canvas vs hi resolution layout exports.

Reply:
it is in pixels (will add that). Using floating point numbers for pixel distance probably wouldn't make sense - interpolating elevation values of neighboring pixels seems wrong (because of discontinuities).

I am wondering if it would make sense to have the distance e.g. in millimeters, but then we would convert it to integer pixel value - it may be weird that small adjustments of the value wouldn't trigger any change in behavior...

Reply:
That's why I'd go the double + unit route. It's what we do elsewhere to handle this, even for situations where the final number must be an integer pixel size (eg font sizes, blur radiuses, etc). Otherwise it'll be impossible to use this effect on print layouts, since the pixel sizes for screen won't be visible at print resolution outputs. (Also there'll be issues sharing projects across hi DPI/non hi DPI systems)